File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq=new PriorityQueue<>();
        for(int i=0;i<nums.length;i++){
            pq.add((long)nums[i]);
        }
        int count=0;
        while(!pq.isEmpty() && pq.size()>=2){
            long first=pq.poll();
            long second=pq.poll();
            if(first <k ){
                long ans=(Math.min(first,second)*2)+Math.max(first,second);
                pq.add(ans);
                count++;
            }else{
                break;
            }
            
        }
        return count;
    }
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.py
from queue import PriorityQueue
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        pq=PriorityQueue()
        for i in nums:
            pq.put(i)
        count=0
        while not pq.empty() and pq.qsize()>1:
            if(pq.queue[0]<k):
                
                ele1=pq.get()
                ele2=pq.get()
                s=min(ele1,ele2)*2 +max(ele1,ele2)
                pq.put(s)
                count=count+1
            else:
                break
        return count






        